exporter/collector/internal/datapointstorage/datapointcache.go

```diff
@@ -212,27 +212,57 @@ func (c *Cache) gc(shutdown <-chan struct{}, tickerCh <-chan time.Time) bool {
 }
 
 // Identifier returns the unique string identifier for a metric.
-func Identifier(resource *monitoredrespb.MonitoredResource, extraLabels map[string]string, metric pmetric.Metric, attributes pcommon.Map) string {
-	var b strings.Builder
+func Identifier(resource *monitoredrespb.MonitoredResource, extraLabels map[string]string, metric pmetric.Metric, attributes pcommon.Map) (uint64, error) {
+	var err error
+	h := fnv.New64()
 
-	// Resource identifiers
-	if resource != nil {
-		fmt.Fprintf(&b, "%v", resource.GetLabels())
+	_, err = h.Write([]byte(resource.GetType()))
+	if err != nil {
+		return 0, err
+	}
+	_, err = h.Write([]byte(metric.Name()))
+	if err != nil {
+		return 0, err
 	}
 
-	// Instrumentation library labels and additional resource labels
-	fmt.Fprintf(&b, " - %v", extraLabels)
-
-	// Metric identifiers
-	fmt.Fprintf(&b, " - %s -", metric.Name())
-	attrsIds := make([]string, 0, attributes.Len())
+	attrs := make(map[string]string)
 	attributes.Range(func(k string, v pcommon.Value) bool {
-		attrsIds = append(attrsIds, k+"="+v.AsString())
+		attrs[k] = v.AsString()
```

Comment:
Why writing to hash directly here?

Reply:
yeah, they aren't guaranteed to be sorted, so we are sorting first.